libbeat/processors/add_docker_metadata/add_docker_metadata.go

```diff
@@ -48,6 +52,18 @@ const (
 var processCgroupPaths = cgroup.ProcessCgroupPaths
 
 func init() {
+	// check if Docker is available in running environment
+	cli, err := client.NewEnvClient()
+	if err != nil {
+		logp.Info("%v: docker environment not detected.", processorName)
+		return
+	}
+	info, err := cli.Info(context.Background())
+	if err != nil {
+		logp.Info("%v: docker environment not detected.", processorName)
+		return
+	}
+	logp.Info("docker environment detected: %v", info)
```

Comment:
I think this code should be called from `New`, as it is done in `add_cloud_metadata`, so the behaviour is that the processor is loaded, but it does nothing.

If we return here before the plugin is registered configurations will fail because `add_docker_metadata` will be an unknown processor.

Reply:
And then you wouldn't need the check when loading the list of processors :slightly_smiling_face: